docs/pipelines.md

```diff
@@ -516,6 +517,203 @@ In particular:
 
 The `description` field is an optional field and can be used to provide description of the `Pipeline`.
 
+## Adding `Finally` to the `Pipeline`
+
+You can specify a list of one or more final tasks under `finally` section. Final tasks are guaranteed to be executed
```

Comment:
It may be worth saying that all tasks specified here will be executed in parallel

Reply:
Added in the latest implementation PR: https://github.com/tektoncd/pipeline/pull/2661/files#diff-db9ac78ce08367ea2b7db8c1412049b6R535